### PR TITLE
DGJ-1131 replace single quote with double single quote while sending data to ODS

### PR DESCRIPTION
**Pull request by bhumin-fw**
## Summary

DGJ-1131 replace single quote with double single quote while sending data to ODS
- Setup dynamic character replace in string value

## Changes
- Add provision in `sendDataToODS` delegates which replace provides characters.
- Added readme file on how to use input param to play with request object to ODS.

## Notes
NA

### CODE REVIEW

File: forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/listeners/SendSubmissionToODSDelegate.java [hunk 1]
```diff
@@ -87,11 +90,15 @@ private void sendSubmissionToODS(DelegateExecution execution) throws IOException
...
-            json = objectMapper.writeValueAsString(values);
+            values = convertToSnake(values);
+        }
+        if (replaceText != null) {
+            String[][] replaceArray = parseStringToArray(replaceText.split("\\s*,\\s*"));
+            if (replaceArray.length > 0) {
+                values = replaceTextAll(values, replaceArray);
+            }
         }
-        System.out.println(json);
```
Comment (iman-jamali-fw) [inline forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/listeners/SendSubmissionToODSDelegate.java:94]: This line proved to be very useful for debugging purposes. I'd leave it here.

File: forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/listeners/SendSubmissionToODSDelegate.java [hunk 2]
```diff
@@ -129,4 +136,39 @@ public Map<String, Object> convertToSnake(Map<String, Object> values) {
...
+        for (int i = 0; i < input.length; i += 2) {
+            if (i+1 <= input.length && init < replaceArraylength) {
+                nestedArray[init][0] = input[i];
+                nestedArray[init][1] = input[i+1];
+                init++;
+            }
+        }
+        return nestedArray;
+    }
 }
```
Comment (iman-jamali-fw) [inline forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/listeners/SendSubmissionToODSDelegate.java:174]: I'm wondering if we change the input type in the workflow from string to an array of strings like ["@, #", "-,+"] makes things easier for the form designer. However, I'm unsure how hard it is to escape ' and " in that case.

File: forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/listeners/SendSubmissionToODSDelegate.java (hunk 1 above)
Comment (bhumin-fw) [reply forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/listeners/SendSubmissionToODSDelegate.java:94]: very next line is that prints json value anyway if you pass debug true. That's why remove additional log here.
```
if(debug) {
    System.out.println("Sending values to ODS: " + json);
}
```

Comment (iman-jamali-fw) [reply forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/listeners/SendSubmissionToODSDelegate.java:94]: Oh, you're right! I thought that on is the one. No worries then

File: forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/listeners/SendSubmissionToODSDelegate.java (hunk 2 above)
Comment (bhumin-fw) [reply forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/listeners/SendSubmissionToODSDelegate.java:157]: @iman-jamali-fw I have moved from string array to list. apply your suggestion it and pushed new code.

Comment (iman-jamali-fw) [reply forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/listeners/SendSubmissionToODSDelegate.java:157]: Looks awsome!

**Review by iman-jamali-fw — APPROVED** [review]
Looks great!

**Pull request merged**